ml-proto/host/main.ml

```diff
@@ -22,11 +22,16 @@ let parse name source =
   let lexbuf = Lexing.from_string source in
   lexbuf.Lexing.lex_curr_p <-
     {lexbuf.Lexing.lex_curr_p with Lexing.pos_fname = name};
-  try Parser.script Lexer.token lexbuf with Error.Error (region, s) ->
+  try Parser.script Lexer.token lexbuf with Script.Syntax (region, s) ->
     let region' = if region <> Source.no_region then region else
       {Source.left = Lexer.convert_pos lexbuf.Lexing.lex_start_p;
        Source.right = Lexer.convert_pos lexbuf.Lexing.lex_curr_p} in
-    raise (Error.Error (region', s))
+    raise (Script.Syntax (region', s))
+
+let error at category msg =
```

Comment:
Would it make sense to make error take (at, msg) as a tuple and a category string to simplify the exception handling pattern match below?